### DISCUSSION TITLE
How to customize the LuaLS configuration? And which configs are even needed with lazydev?

### DISCUSSION THREAD

**Discussion by miguno** [folke/lazydev.nvim#31] [Q&A] [answered]
When I was still using neodev, I explicitly installed `lua_ls` via mason.

This allowed me to configure LuaLS like so:

```lua
local servers = {
  lua_ls = {
    settings = {
      Lua = {
          version = "LuaJIT",
          path = lua_runtime_path,
        },
        diagnostics = {
          enable = true,
          -- Get the language server to recognize the globals like `vim`
          globals = { "vim", "describe" },
        },
        workspace = {
          library = vim.api.nvim_get_runtime_file("lua", true),
          checkThirdParty = false,
        },
        -- Do not send telemetry data containing a randomized but unique identifier
        telemetry = {
          enable = false,
        },
      },
    },
  },
  -- ... other LSPs ...
}
```

I have two questions here:

1. Which of the above settings are automatically set by lazydev? For example, I suppose that the workspace is automatically set up correctly?
    - I am asking because, while migrating from neodev.nvim to layzdev.nvim, I observed that layzdev would automatically add its own instance of LuaLS, so I ended up having two instances running. (I subsequently removed `lua_ls` from my mason setup, so now there is just one.) This behavior wasn't obvious to me when I read lazydev's README, so I am wondering whether there is further auto-magic happening.
3. For the settings that still need to be added manually when using lazydev, how can I do that? For example, how can I disable LuaLS' telemetry?

**Comment by folke** [comment, marked as answer]
First of all, lazydev does **NOT** install **LuaLS**.
If you mean you see two progress bars, then that is to be expected and has always been the case.
That's LuaLS loading your workspace and its fallback scope.

1. https://github.com/folke/lazydev.nvim/blob/main/lua/lazydev/workspace.lua#L173
2. nothing special. You still should setup LuaLS as you did before. lazydev doesn't do that for you.

**Reply by miguno** [reply]
Thanks for the quick reply and for your work on lazydev and all your other plugins!
